### PR TITLE
Mod toolkit, the forgotten stage: DM command

### PR DESCRIPTION
**Pull request by Akarys42**
*I may or may not have forgotten about this command during the last PR*

This PR adds a DM command reserved for moderators and SC members. It also fixes a major bug causing the error handler to just not work.

### CODE REVIEW

File: bot/exts/moderation/mod_utils.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from bot.bot import Bot
+from bot.constants import Roles
+from discord import Forbidden, Member
+from discord.ext.commands import Cog, Context, command, has_any_role
+from loguru import logger
+
+
+class ModUtils(Cog):
+    """Cog used for various utility moderation commands."""
```
Comment (gustavwilliam) [inline bot/exts/moderation/mod_utils.py:9]:
```suggestion
    """Cog used for various moderation utility commands."""
```
I feel like this sounds better.

File: bot/exts/moderation/mod_utils.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        super().__init__()
+
+    @command(aliases=("dm",))
+    @has_any_role(Roles.moderators, Roles.steering_council)
+    async def send_dm(self, ctx: Context, user: Member, *, message: str) -> None:
+        """Send a DM to the specified user."""
+        logger.info(f"Sending message {message!r} to {user}.")
+        try:
+            await user.send(message)
+        except Forbidden:
```
Comment (gustavwilliam) [inline bot/exts/moderation/mod_utils.py:22]: Should we log this as well, so we have a record of the fact that the DM arrived or didn't arrive?

File: bot/exts/moderation/mod_utils.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            await ctx.message.add_reaction("\N{CROSS MARK}")
+        else:
+            await ctx.message.add_reaction("\N{OK HAND SIGN}")
+
+
+def setup(bot: Bot) -> None:
+    """Load the moderation log during setup."""
```
Comment (gustavwilliam) [inline bot/exts/moderation/mod_utils.py:29]:
```suggestion
    """Load the moderation utils cog during setup."""
```
Something like this?

**Review by gustavwilliam — CHANGES_REQUESTED** [review]
Thanks for the fast PR! I just have a few minor comments.

File: bot/exts/moderation/mod_utils.py (hunk 2 above)
Comment (Akarys42) [reply bot/exts/moderation/mod_utils.py:22]: We do get a reaction on the message depending on whenever it failed or not

File: bot/exts/backend/error_handler.py [hunk 4]
```diff
@@ -146,11 +144,12 @@ async def handle_unexpected_error(
                 logger.exception("Fetch failed", exc_info=discord_exc)
                 return
 
-        await dev_alerts.send(embed=push_alert)
+        # Trigger the logger before trying to use Discord in case that's the issue
         logger.error(
             f"Error executing command invoked by {ctx.message.author}: {ctx.message.content}",
             exc_info=error,
         )
+        await dev_alerts.send(embed=push_alert)
```
Comment (gustavwilliam) [inline bot/exts/backend/error_handler.py:152]: Good thinking!

**Review by gustavwilliam — APPROVED** [review]
LGTM! (Code review only)

File: bot/exts/moderation/logs.py [hunk 5]
```diff
@@ -8,6 +8,24 @@
 from loguru import logger
 
 
+def get_post_message(bot: Bot) -> Callable:
+    """Magic function returning the `post_message` function."""
+    cog = bot.get_cog("ModerationLog")
+    if not cog:
+        raise ValueError("The ModerationLog cog can't be found.")
+
+    return cog.post_message
```
Comment (codephile1221) [inline bot/exts/moderation/logs.py:17]: What's this for? I don't see it used anywhere.

**Review by codephile1221 — CHANGES_REQUESTED** [review]
Looks good :+1: .

File: bot/exts/moderation/logs.py (hunk 5 above)
Comment (Akarys42) [reply bot/exts/moderation/logs.py:17]: Right now it isn't but I want it to be part of the public API of this cog, I bet it will be used in the future

Comment (codephile1221) [reply bot/exts/moderation/logs.py:17]: Good call.

**Review by codephile1221 — APPROVED** [review]
Looks good to me!
⭐

**Pull request merged**